Update Featured Content Module styling

Scotchester opened the pull request:
This PR implements several desired styling updates for Featured Content Modules. See https://GHE/CFGOV/platform/issues/3087 for background.

/cc @designlanguage @keelerr

## Removals

- Remove category headers from FCM templates

## Changes

- Change FCM min height from 320px to 220px
- Set FCM image width to 270px, let text flex
- Change FCM breakpoint to 600px
- Add bottom padding to FCM image on XS screens
- Sets border on the FCM itself, instead of surrounding block
  - The border is inherent to the FCM styling and shouldn't be set on the block. It makes even less sense now that the block modifier for adding the negative side margins is no longer used.
- Updates docs

## Testing

1. Check out a working branch
1. Run `gulp docs`
1. Go into the documentation directory with `cd docs`
1. Run `bundle exec jekyll serve` and visit http://localhost:4000/components/cf-layout/#featured-content-module
1. Go into responsive mode and view the FCM at various window widths

## Screenshots

### Before

![Screen Shot 2019-06-06 at 13 12 40](https://user-images.githubusercontent.com/1044670/59052375-fc228400-885c-11e9-8f16-428041517f46.png)

### After

![Screen Shot 2019-06-06 at 13 13 33](https://user-images.githubusercontent.com/1044670/59052395-047abf00-885d-11e9-9005-d5a6290cc6ef.png)

## Todos

- [ ] Open corresponding PR in cfgov-refresh

## Checklist

- [x] PR has an informative and human-readable title
- [x] PR has been tagged with the type of change it includes
      (breaking, enhancement, bug, documentation or internal)
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code can be automatically merged (no conflicts)
- [x] Code follows the standards laid out in the [development playbook](https://github.com/cfpb/development)
- [x] Passes all existing automated tests
- [x] Any _change_ in functionality is tested
- [x] New functions are documented (with a description, list of inputs, and expected output)
- [x] Placeholder code is flagged / future todos are captured in comments
- [x] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [x] Project documentation has been updated in this PR with the `gulp docs` command
- [x] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

## Testing checklist

### Browsers

- [x] Chrome
- [x] Firefox
- [x] Safari
- [x] Internet Explorer 8, 9, 10, and 11
- [x] Edge
- [x] iOS Safari
- [x] Chrome for Android

### Accessibility

- [x] Keyboard friendly
- [x] Screen reader friendly

### Other

- [x] Is useable without CSS
- [x] Is useable without JS
- [x] Flexible from small to large screens
- [x] No linting errors or warnings
- [x] JavaScript tests are passing

anselmbradford commented on docs/_includes/usage/cf-layout/usage.md:
```diff
@@ -1290,208 +1290,181 @@ height of the next parent using `position: relative;`. This means that the
+an optional call to action, and a static or video visual.
+They are intended to be used in a main content column next to a sidebar.
 
 Text is full width & displayed above the visual in the default/mobile view.
```
Minor nitpick, but why are we using ampersands here? Probably would be stylistically better to spell out `and`.

Scotchester replied:
:+1: Updated in 43458fae.

anselmbradford commented on docs/_includes/usage/cf-layout/usage.md:
```diff
@@ -1290,208 +1290,181 @@ height of the next parent using `position: relative;`. This means that the
+At larger screen sizes, the image moves to the right of the text,
+occupying a fixed width of 270px (equal to 3 of 12 columns at max page width).
+When creating a static image for the visual, it should be 1076px × 606px
+(a 16:9 aspect ratio), which is 2x the rendered with at the maximum size
```
`which is 2x the rendered with` … should be `width` not `with`.

Scotchester replied:
Good catch; thanks! Fixed in 43458fae.

anselmbradford commented on docs/_includes/usage/cf-layout/usage.md:
```diff
@@ -1290,208 +1290,181 @@ height of the next parent using `position: relative;`. This means that the
+(a 16:9 aspect ratio), which is 2x the rendered with at the maximum size
+at which the visual will be seen (on a 600px wide display).
+
+By default, the image is anchored on its left when the FCM is viewed on a
```
Should it say `anchored on its center-left` or `anchored on its middle-left`, as that's what it looks like it is doing?

Scotchester replied:
I tried to clarify this below in the right-anchored description (and I should have written it the same way here), but it looks like that was confusing to you also, so perhaps I should go back to the drawing board 😆 

How does this sound (including some minor edits to the previous paragraph)?

---

Text is full width and displayed above the visual in the default/mobile view.
When creating a static image for the visual, it should be 1076px × 606px
(a 16:9 aspect ratio), which is 2x the rendered width at the maximum size
at which the visual will be seen (on a 600px wide display).

At larger screen sizes, the image moves to the right of the text,
occupying a fixed width of 270px (equal to 3 of 12 columns at max page width).
By default, the left edge of the image is anchored to the left side of the
visual's 270px-wide area, and the right side is cropped off.
The height of the visual area decreases as screen size increases,
resulting in slightly different image cropping at different screen sizes,
but the left edge of the image remains anchored in view.
See below for modifiers that change the image anchoring.

Scotchester replied:
Updated in 43458fae.

anselmbradford replied:
> The height of the visual area decreases as screen size increases,
resulting in slightly different image cropping at different screen sizes,
but the left edge of the image remains anchored in view.

Still not totally clear how the cropping works without looking at it closer. The top AND bottom get cropped as screen size increases? Or is it anchored to upper-left and the bottom only gets cropped?

anselmbradford replied:
Also, should it be `Text is full-width …`? Not sure what we use elsewhere.

Scotchester replied:
> Still not totally clear how the cropping works without looking at it closer. The top AND bottom get cropped as screen size increases? Or is it anchored to upper-left and the bottom only gets cropped?

The full height of the image remains visible; top and bottom are not cropped at all. I can add that in.

This is extraordinarily difficult to describe in words. The best thing to do is open the responsive design mode and play around with it. Maybe I should literally say that in the docs, too :)

> Also, should it be `Text is full-width …`? Not sure what we use elsewhere.

This phrase is leftover from the previous iteration of the docs. It is probably assumed to be true when one is talking about the mobile view, so it may not be necessary to state it that way. How about... "The text and visual are stacked vertically at screen widths of 600px and under."?

anselmbradford commented on docs/_includes/usage/cf-layout/usage.md:
```diff
@@ -1290,208 +1290,181 @@ height of the next parent using `position: relative;`. This means that the
-</div>
+### Anchoring the visual on the right
+
+When one wants to anchor a featured content module's visual on the right
```
Could be shortened to `To anchor a featured content module's visual on the right…`

Scotchester replied:
:+1: Updated in 43458fae.

anselmbradford commented on docs/_includes/usage/cf-layout/usage.md:
```diff
@@ -1290,208 +1290,181 @@ height of the next parent using `position: relative;`. This means that the
+(for example, when displaying a map with a copyright watermark on the right),
+add the `o-featured-content-module__right` modifier class
+to the `o-featured-content-module` organism.
+This anchors the right side of the image to the right of the available space
```
Should it be `right-middle` or `right-center`?

anselmbradford commented on docs/_includes/usage/cf-layout/usage.md:
```diff
@@ -1290,208 +1290,181 @@ height of the next parent using `position: relative;`. This means that the
-</div>
+At larger screen sizes, the image moves to the right of the text,
+occupying a fixed width of 270px (equal to 3 of 12 columns at max page width).
+When creating a static image for the visual, it should be 1076px × 606px
```
Says image should be 1076×606, but the example images used are 540×270. Should this be noted?

Scotchester replied:
Updated all the placeholder images to be 1076x606. This also conveniently gives us an image where the kitten is more centered, which makes the different anchoring more obvious.

anselmbradford commented on packages/cf-layout/src/organisms/featured-content-module.less:
```diff
@@ -19,25 +19,31 @@
         padding-bottom: unit( @grid_gutter-width / @base-font-size-px, em );
 
+        .respond-to-max( @bp-xs-max, {
+            padding-right: unit( ( @grid_gutter-width / 2 ) / @base-font-size-px, em );
```
TRIVIAL: This and the line below have extra parens you don't need. `unit( @grid_gutter-width / 2 / @base-font-size-px, em );` should work.

anselmbradford commented on packages/cf-layout/src/organisms/featured-content-module.less:
```diff
@@ -19,25 +19,31 @@
+        .respond-to-max( @bp-xs-max, {
+            padding-right: unit( ( @grid_gutter-width / 2 ) / @base-font-size-px, em );
+            padding-bottom: unit( ( @grid_gutter-width / 2 ) / @base-font-size-px, em );
+            padding-left: unit( ( @grid_gutter-width / 2 ) / @base-font-size-px, em );
```
Three lines with extra parens. I'll leave it up to you if you want to remove them. We have both formats in the wild I think.

Scotchester commented:
@anselmbradford Thanks for the review! I believe I have addressed all your feedback. Ready for re-review.

P.S.: I still want to do a little more cross-browser testing. I see an issue in IE9 already 😩

anselmbradford commented on packages/cf-layout/src/cf-layout.less:
```diff
@@ -45,8 +45,8 @@
-@fcm-bp-max:                    700px;
-@fcm-bp-min:                    701px;
+@fcm-bp-max:                    600px;
+@fcm-bp-min:                    601px;
```
Do we need these any more or could we use the standard breakpoints directly? https://github.com/cfpb/capital-framework/blob/ad6871887c3d851b6dfe34fd6dd36cf63e27c843/packages/cf-core/src/cf-vars-breakpoints.less#L9-L10

Scotchester replied:
Talked to Chris to ensure he didn't intend something that I was unaware of. He did not, so I cleaned up the variables.

anselmbradford commented on packages/cf-layout/src/organisms/featured-content-module.less:
```diff
@@ -19,25 +19,31 @@
         .respond-to-min( @fcm-bp-min, {
-            padding-right: unit( @grid_gutter-width / @base-font-size-px, em );
+            flex: 1;
+            padding-right: 300px;
```
Just checking, fixed padding here is correct?

Scotchester replied:
It is, but it should be calculated from variables. Updated in the forthcoming commit.

Scotchester commented:
I think this is good to go. Fixed the IE9 bug by adding an explicit `top: 0;` to the visual's absolute positioning. @anselmbradford, any final comments?
